pkg/resources/digitalocean/cloud.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package digitalocean
+
+import (
+	"errors"
+	"os"
+
+	"github.com/digitalocean/godo"
+	"golang.org/x/oauth2"
+	"k8s.io/kops/pkg/resources/digitalocean/dns"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+)
+
+type TokenSource struct {
+	AccessToken string
+}
+
+func (t *TokenSource) Token() (*oauth2.Token, error) {
+	token := &oauth2.Token{
+		AccessToken: t.AccessToken,
+	}
+	return token, nil
+}
+
+type Cloud struct {
+	client *godo.Client
+
+	Region string
+	tags   map[string]string
+}
+
+func NewCloud() (*Cloud, error) {
+	accessToken := os.Getenv("DO_ACCESS_TOKEN")
+	if accessToken == "" {
+		return nil, errors.New("DO_ACCESS_TOKEN is required")
+	}
+
+	tokenSource := &TokenSource{
+		AccessToken: accessToken,
+	}
+
+	oauthClient := oauth2.NewClient(oauth2.NoContext, tokenSource)
+	client := godo.NewClient(oauthClient)
```

Comment:
The `godo` client is pretty straight forward since it only needs to initialize a http client so there's really no errors to return. See implementation here: https://github.com/digitalocean/godo/blob/master/godo.go#L150-L179